### DISCUSSION TITLE
Nushell webserver library

### DISCUSSION THREAD

**Discussion by ghost** [nushell/nushell#13423] [Show and tell]
Hi,

i had the wish to handle http-requests using nushell functions quite often.  
Now ive sat down and created a library to make this possible (and easy?).

Example usage:

```nu
use webserver.nu *

start_mapped_webserver 8080 {
  "/time.json": {|request|
    format_http 200 "application/json" (date now | to json)
  }
  "/hello.txt": {|request|
    format_http 200 "text/plain" $"Hello ($request.params.name? | default World)!"
  }
}
```

More useful use-cases for which ill use it (probably):
* generate statistics page based on json data
* htop website
* remote control server (start/stop services, etc)

link: <https://github.com/Jan9103/webserver.nu>

**Comment by fdncred** [comment]
This looks cool. Thanks. Feel free to submit a PR to our awesome-nu repo with it.

Maybe related - @cablehead was doing some http request stuff too with https://github.com/cablehead/nu_plugin_from_sse

**Reply by cablehead** [reply]
woah this cool. 🙏 @fdncred . This is a bit more related: https://github.com/cablehead/nu_plugin_http but that's still very much a WIP (and Unix socket specify as that's my current need, but I will add TCP).

For the `h. serve` plugin command I need to add support to Nushell proper for more than one plugin to engine call for a given engine to plugin call. It can currently it can only serve one request at a time 😆, but it's working as a proof of concept.
